cpp/include/cudf/ast/linearizer.hpp

```diff
@@ -0,0 +1,233 @@
+/*
```

Comment:
I would rename this header. Maybe `ast.hpp`? As far as a user is concerned, they don't know or care about what the "linearizer" is. This header just defines the interfaces for defining the user-facing AST representation.

Reply:
This appears to be a remaining TODO. Either in this PR or in a follow up.